tools/development/gstPrototxt.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python3
+##
+# SPDX-License-Identifier: LGPL-2.1-only
+#
+# Prototxt <--> GStreamer pipeline converter
+# Copyright (c) 2020 Samsung Electronics
+#
+##
+# @file   gstPrototxt.py
+# @brief  Prototxt to/from GStreamer pipeline converter
+# @author MyungJoo Ham <myungjoo.ham@samsung.com>
+# @date   30 Jun 2020
+# @bug    No known bugs
+# @todo   WIP
+#
+
+## @todo Get python pbtxt parser
+
+## @todo property link up tensor-repo-*
+
+declaredNames = [] # list of names declared by user.
+generatedNameCounter = 0
+generatedNamePrefix = '__id'
+
+## @brief A class for a filter (object == filter instance)
+class Filter:
+    element = None    # element name (string)
+    name = None       # name property (string)
+    nameGenerated = False   # boolean
+    properties = [] # list of (key, value)
+    src = []        # list of (Filter, string(name of src), string(src-padname. maybe None))
+    sink = []        # list of (Filter, string(name of sink), string(sink-padname. maybe None))
+
+    def __init__(self, elementname, _name):
+        global generatedNameCounter
+        global generatedNamePrefix
+        global declaredNames
+
+        self.element = elementname
+        if _name is not None and len(_name) > 0:
+            name = _name
+            if name in declaredNames:
+                raise RuntimeError("Duplicated name in elements: " + str(name))
+            declaredNames.append(str(name))
+        else:
+            while ((generatedNamePrefix + str(generatedNameCounter)) not in declaredNames):
+                generatedNameCounter = generatedNameCounter + 1
+            name = generatedNamePrefix + str(generatedNameCounter)
+            nameGenerated = True
```

Comment:
bring this outside the `if else` and add this name to `declaredNames`?

Reply:
No, the intention is to manage filters with its id declared (```filtername id=THENAME```) and filters without such (generated) differently. If it is generated (not declared), we can hide the id in GUI as well. If it is declared, we need to show the id in GUI.